frontend/src/components/Forms/MineRecordForm.js

```diff
@@ -3,31 +3,64 @@ import PropTypes from "prop-types";
 import { connect } from "react-redux";
 import { compose } from "redux";
 import { Field, reduxForm, FieldArray, getFormValues } from "redux-form";
-import { Form, Button, Col, Row, Popconfirm, Icon, Collapse } from "antd";
+import { Form, Button, Col, Row, Popconfirm, Icon, Collapse, notification, Tag } from "antd";
+import _ from "lodash";
 import * as FORM from "@/constants/forms";
+import * as Strings from "@/constants/strings";
+import CustomPropTypes from "@/customPropTypes";
 import { required, maxLength, minLength, number, lat, lon } from "@/utils/Validate";
-import { resetForm } from "@/utils/helpers";
 import { renderConfig } from "@/components/common/config";
+import { getCurrentMineTypes } from "@/selectors/mineSelectors";
+import {
+  getConditionalDisturbanceOptionsHash,
+  getConditionalCommodityOptions,
+  getDisturbanceOptionHash,
+  getCommodityOptionHash,
+  getMineStatusOptions,
+  getMineRegionOptions,
+  getMineTenureTypes,
+  getMineTenureTypesHash,
+} from "@/selectors/staticContentSelectors";
 
 const propTypes = {
   handleSubmit: PropTypes.func.isRequired,
   closeModal: PropTypes.func.isRequired,
   change: PropTypes.func.isRequired,
+  handleDelete: PropTypes.func.isRequired,
   title: PropTypes.string,
-  mineStatusOptions: PropTypes.array.isRequired,
-  mineRegionOptions: PropTypes.array.isRequired,
-  mineTenureTypes: PropTypes.array.isRequired,
-  mine_types: PropTypes.array,
-  conditionalDisturbanceOptions: PropTypes.object.isRequired,
-  conditionalCommodityOptions: PropTypes.object.isRequired,
-  initialValues: PropTypes.object,
+  mineStatusOptions: CustomPropTypes.options.isRequired,
+  mineRegionOptions: CustomPropTypes.options.isRequired,
+  mineTenureTypes: CustomPropTypes.options.isRequired,
+  mineCommodityOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  mineDisturbanceOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  mine_types: PropTypes.arrayOf(CustomPropTypes.mineTypes),
+  mineTenureHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  conditionalDisturbanceOptions: PropTypes.objectOf(PropTypes.arrayOf(CustomPropTypes.option))
+    .isRequired,
+  conditionalCommodityOptions: PropTypes.objectOf(PropTypes.arrayOf(CustomPropTypes.option))
+    .isRequired,
+  currentMineTypes: PropTypes.arrayOf(CustomPropTypes.mineTypes),
+};
+
+const defaultProps = {
+  title: "",
+  currentMineTypes: [],
+  mine_types: [],
 };
 
 export class MineRecordForm extends Component {
   state = {
     activeKey: [],
+    usedTenureTypes: [],
   };
 
+  componentDidMount() {
+    const existingMineTypes = this.props.currentMineTypes
```

Comment:
You have a default prop for `currentMineTypes`, so it should never be `undefined`. Mapping over an empty array will return an empty array. So, this should work just fine:
```javascript
const existingMineTypes = this.props.currentMineTypes.map(({ mine_tenure_type_code }) => mine_tenure_type_code)
this.setState({ usedTenureTypes: existingMineTypes });
```

It's also possible to use the lodash `map()` function to reduce the boilerplate, if you would like to:
```javascript
// map() is overloaded to behave like .pluck() from other libraries
const existingMineTypes = map(this.props.currentMineTypes, "mine_tenure_type_code")
this.setState({ usedTenureTypes: existingMineTypes });
```

Reply:
much cleaner